### PR TITLE
Add product page link and move download in order details

### LINKED ISSUE #24702: Link to products in order details for client
#### Is your feature request related to a problem?

In order details in customer account page product section only has name and reference number. Would be nice to also have link to product page there to see what it actually is.

#### Describe the solution you'd like

Name of product, when inspected already is an `a` element, but is missing `href`, looks like feature already was planned, did not make it in its entirety.

### PR DESCRIPTION
**Pull request by marekjedrzejewski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Add product page link in order details into name on product name and download link as separate thing.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #24702
| How to test?      | Go to order page from customer account.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24730)

### CODE REVIEW

**Review by NeOMakinG — DISMISSED** [review]
LGTM

**Review by PierreRambaud — DISMISSED** [review]

File: themes/classic/templates/customer/_partials/order-detail-no-return.tpl [hunk 1]
```diff
@@ -37,13 +37,16 @@
...
-              <a {if isset($product.download_link)}href="{$product.download_link}"{/if}>
+              <a href="{$urls.pages.product}&id_product={$product.id_product}">
                 {$product.name}
               </a>
             </strong><br/>
             {if $product.product_reference}
               {l s='Reference' d='Shop.Theme.Catalog'}: {$product.product_reference}<br/>
             {/if}
+            {if isset($product.download_link)}
+              <a href="{$product.download_link}">{l s='download' d='Shop.Theme.Catalog'}</a><br/>
```
Comment (Julievrz) [inline themes/classic/templates/customer/_partials/order-detail-no-return.tpl]:
```suggestion
              <a href="{$product.download_link}">{l s='Download' d='Shop.Theme.Actions'}</a><br/>
```

**Review by kpodemski — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by florine2623** [comment]
LGTM

Thanks @marekjedrzejewski !

**Comment by Progi1984** [comment]
Thanks @marekjedrzejewski & @florine2623

**Pull request merged**